swagger-config/marketing/ruby/templates/api_client.mustache

```diff
@@ -57,7 +57,12 @@ module {{moduleName}}
       headers[:Authorization] = "Bearer #@access_token" if @is_oauth
 
       host = @server.length > 0 ? @host.sub('server', @server) : @host
-      conn = Excon.new(host + path, :headers => headers, :read_timeout => @read_timeout, :write_timeout => @write_timeout, :connect_timeout => @connect_timeout)
+            conn = Excon.new(host + path,
+                       :headers => headers,
+                       :read_timeout => @read_timeout,
+                       :write_timeout => @write_timeout,
+                       :connect_timeout => @connect_timeout,
+                       :middlewares => [Excon.defaults[:middlewares], Excon::Middleware::Decompress].flatten)
```

Comment:
This is the magic line. It sets the middlewares used by the new Excon connection to be the defaults (`Excon.defaults[:middlewares]`) and `Excon::Middleware::Decompress`, which knows how to decompress gzip response data.